File: src/core/get_alias.nim
import tables
import core/session_state
import models/command

proc getAlias*(cmd: Command, state: SessionState): Command =
  if state.aliases.hasKey(cmd.arguments[0]):
    var alias = state.aliases[cmd.arguments[0]]
    if cmd.arguments.len > 1:
      let remainingArgs = cmd.arguments[1 .. ^1]
      alias.arguments = alias.arguments & remainingArgs
    return alias
  return ("", @[])

File: src/core/get_alias.rs
use crate::models::command::Command;
use crate::core::session_state::SessionState;

pub fn get_alias(cmd: &Command, state: &SessionState) -> Option<Command> {
    if let Some(alias) = state.aliases.get(&cmd.arguments[0]) {
        let mut command = alias.clone();
        if cmd.arguments.len() > 1 {
            let remaining_args = &cmd.arguments[1..];
            command.arguments.extend_from_slice(remaining_args);
        }
        Some(command)
    } else {
        None
    }
}
